File: javascript/algoritmos_busca/binarySearch.c

#include <stdio.h>

void binarySearch(int value, int array[], int size) {
    int first = 0;
    int last = size - 1;
    int found = 0;

    while(first <= last) {
        int middle = (first + last) / 2;
        if(array[middle] == value) {
            found = 1;
            printf("Value %d found at position %d ", value, middle);
            break;
        }else if(array[middle] > value) {
            last = middle - 1;
        }else{
            first = middle + 1;
        }
    }
    if(!found) {
        printf("Value %d not found \n", value);
    }
}

int main() {
    int orderedArray[] = {1, 3, 5, 6, 8, 12, 13, 15, 16};
    int valueToFind = 15;
    int size = sizeof(orderedArray) / sizeof(orderedArray[0]);
    binarySearch(valueToFind, orderedArray, size);
}

File: leetcode/binarySearch.js

const binarySearch = function(array, target) {
    let begin = 0;
    let end = array.length - 1;
    let found = false;

    while(begin <= end) {
        const middle = (begin + end) / 2;
        if(array[middle] === target) {
            found = true;
            console.log("Element " + target + " found in position " + middle);
            break;
        }else if(array[middle] <= target) {
            end = middle + 1;
        }else{
            begin = middle - 1;
        }
    }
    if(!found) {
        console.log("Not found");
    }
} 

const array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
const target = 99;

const result = binarySearch(array, target);
console.log(result)